src/Maestro/Maestro.Web/Helpers.cs

```diff
@@ -99,5 +102,87 @@ public static async Task<IActionResult> ProxyRequestAsync(this HttpContext conte
+            // summarize on the definition id and ultimately choose the definition that took the longest.
+            string commonQueryText = @"| where Result != 'failed' and Result != 'canceled' 
+                | where FinishTime > ago(_Days) 
+                | extend duration = FinishTime - StartTime 
+                | summarize average_duration = avg(duration) by DefinitionId";
+
+            // We only want the pull request time from the public ci. We exclude on target branch,
+            // as all PRs come in as refs/heads/#/merge rather than what branch they are trying to
+            // apply to.
+            string publicQueryText = $@"TimelineBuilds 
+                | project Repository, SourceBranch, TargetBranch, DefinitionId, StartTime, FinishTime, Result, Project, Reason
+                | where Project == '{prProject}'
+                | where Repository endswith _Repository
+                | where Reason == 'pullRequest' 
+                | where TargetBranch == _SourceBranch
+                {commonQueryText}";
+
+            // For the official build times, we want the builds that were generated as a CI run 
+            // (either batchedCI or individualCI) for a specific branch--i.e. we want the builds
+            // that are part of generating the product.
+            string internalQueryText = $@"TimelineBuilds 
+                | project Repository, SourceBranch, DefinitionId, StartTime, FinishTime, Result, Project, Reason
+                | where Project == 'internal' 
+                | where Repository endswith _Repository
+                | where Reason == 'batchedCI' or Reason == 'individualCI'
+                | where SourceBranch == _SourceBranch
+                {commonQueryText}";
+
+            Dictionary<string, KustoQuery> queries = new Dictionary<string, KustoQuery>();
+            queries["internal"] = new KustoQuery(internalQueryText, parameters);
+            queries["public"] = new KustoQuery(publicQueryText, parameters);
+
+            return queries;
+        }
+
+        public static Tuple<int, TimeSpan> ParseBuildTime(IDataReader reader)
+        {
+            // There was an exception when we queried the database
+            if (reader == null)
+            {
+                return null;
+            }
+            Dictionary<int, TimeSpan> buildTimeResults = new Dictionary<int, TimeSpan>();
+
+            while (reader.Read())
+            {
+                int definitionId = Int32.Parse(reader.GetString(0));
+                TimeSpan duration = (TimeSpan) reader.GetValue(1);
+                buildTimeResults[definitionId] = duration;
+            }
+
+            // There were no results
+            if (buildTimeResults.Count() == 0)
+            {
+                return null;
+            }
+
+            int maxDefinitionId = buildTimeResults.Aggregate((l,r) => l.Value > r.Value ? l : r).Key;
```

Comment:
>maxDefinitionId [](start = 16, length = 15)

Why save only the Key... to turn around and then re-look it up? Save the whole pair and return it. :-)